pennylane/transforms/decompositions/single_qubit_unitary.py

```diff
@@ -20,16 +20,17 @@
 
 
 def _convert_to_su2(U, return_global_phase=False):
-    r"""Convert a 2x2 unitary matrix to :math:`SU(2)`.
+    r"""Convert a 2x2 unitary matrix to :math:`SU(2)`. (batched operation)
```

Comment:
I see why you replicated this function in `controlled_decompositions.py`. I think a better solution would be to reuse this function and wrap the input matrix in an extra dimension and then access the first item inside the output to get the SU(2) matrix.

```pycon
>>> U = qml.PauliX.compute_matrix()
>>> U_su2 = _convert_to_su2([U])[0]
array([[0.000000e+00+0.j, 6.123234e-17-1.j],
       [6.123234e-17-1.j, 0.000000e+00+0.j]])
```

This way you can reuse the function without errors and not have an extra dimension in the output.

Reply:
I made the new `_convert_to_su2` use the old one, better?